File: database-driver-plugin/src/main/java/de/lino/database/database/sql/microsoft/MicrosoftSQLServerDatabaseProvider.java
package de.lino.database.database.sql.microsoft;

import de.lino.database.database.auth.Credentials;
import de.lino.database.database.DatabaseType;
import de.lino.database.database.sql.SQLDatabaseProvider;
import de.lino.database.database.sql.SQLExecution;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * The {@link SQLDatabaseProvider} for Microsoft SQL Server, connected via {@link SQLExecution}
 * using {@link DatabaseType#MICROSOFT_SQL_SERVER}'s JDBC driver and URL scheme.
 */
@Getter
public class MicrosoftSQLServerDatabaseProvider extends SQLDatabaseProvider {

    /**
     * Connects to a Microsoft SQL Server database with {@code credentials}.
     *
     * @param credentials the login credentials and connection details to connect with
     */
    public MicrosoftSQLServerDatabaseProvider(@NotNull Credentials credentials) {
        super(DatabaseType.MICROSOFT_SQL_SERVER, new SQLExecution(DatabaseType.MICROSOFT_SQL_SERVER, credentials));
    }
    
}

File: python/database-driver-plugin/src/database_driver/plugin/database/sql/microsoft/microsoft_sql_server_database_provider.py
"""Mirror of ``de.lino.database.database.sql.microsoft.MicrosoftSQLServerDatabaseProvider``."""

from __future__ import annotations

from database_driver.api.database.auth.credentials import Credentials
from database_driver.api.database.database_type import DatabaseType
from database_driver.plugin.database.sql.sql_database_provider import SQLDatabaseProvider
from database_driver.plugin.database.sql.sql_execution import SQLExecution


class MicrosoftSQLServerDatabaseProvider(SQLDatabaseProvider):
    """The ``SQLDatabaseProvider`` for Microsoft SQL Server, connected via ``SQLExecution`` using
    ``DatabaseType.MICROSOFT_SQL_SERVER``'s driver and connection scheme."""

    def __init__(self, credentials: Credentials) -> None:
        super().__init__(DatabaseType.MICROSOFT_SQL_SERVER, SQLExecution(DatabaseType.MICROSOFT_SQL_SERVER, credentials))
